src/services/PackageService.ts

```diff
@@ -0,0 +1,56 @@
+import { GitHubClient as client } from "@/clients/GithubClient.ts";
+import type { IAsset } from "@/lib/types";
+import { UAParser } from "ua-parser-js";
+
+class PackageService {
+  async getRankedPackages(user: string, repo: string, userAgent: string) {
+    const latestRelease = await client.getLatestRelease(user, repo);
+    const parsedUA = UAParser(userAgent);
+    const rankedPackages = {
+      ranked_packages: this.rankPackages(latestRelease.assets, parsedUA).sort(
+        (a, b) => {
+          const bHasOS = b.matches.exact_match.includes("OS");
+          const aHasOS = a.matches.exact_match.includes("OS");
+          return (
+            b.score - a.score ||
+            ((bHasOS && !aHasOS) || (!bHasOS && aHasOS) ? 1 : -1) ||
+            b.downloadCount - a.downloadCount
+          );
+        },
+      ),
+    };
+    return { ...rankedPackages, ...latestRelease };
```

Comment:
lepiej zamiast destructurować będzie chyba po prostu zwrócić obiekt z tymi propami?
```suggestion
    return { rankedPackages, latestRelease };
```

Reply:
Tak, jedyne co to jeszcze assets wywalę z oryginalnego przedmiotu bo to duplikaty niepotrzebne. Znalazłem dokumentację o destrukturyzacji dzisiaj xD